api/src/db/models/user_models.py

```diff
@@ -113,6 +113,7 @@ class UserSavedOpportunity(ApiSchemaTable, TimestampMixin):
     opportunity: Mapped[Opportunity] = relationship(
         "Opportunity", back_populates="saved_opportunities_by_users"
     )
+    is_deleted: Mapped[bool] = mapped_column(nullable=True)
```

Comment:
Use the python typing instead, otherwise I'm not sure what the behavior is if the value was null in the DB and it tries to pull it in
```suggestion
    is_deleted: Mapped[bool | None]
```